saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/MultilayerPerceptron.java

```diff
@@ -0,0 +1,453 @@
+        switch (activation_function){
+            case SIGMOID:
+                return sigmoid(x);
+            case TANH:
+                return tanh(x);
+            case RELU:
+                return relu(x);
+            case LEAKY_RELU:
+                return leaky_relu(x);
+            case SOFTPLUS:
+                return softplus(x);
+            default:
+                System.out.println("Invalid activation function");
+                return 0;
+        }
+    }
+
+    //apply the derivative of the activation function, given the argument and the type of activation function to be used
+    private double apply_activation_derivative(double x, int activation_function){
+        switch (activation_function){
+            case SIGMOID:
+                return sigmoid_derivative(x);
+            case TANH:
+                return tanh_derivative(x);
+            case RELU:
+                return relu_derivative(x);
+            case LEAKY_RELU:
+                return leaky_relu_derivative(x);
+            case SOFTPLUS:
+                return softplus_derivative(x);
+            default:
+                System.out.println("Invalid activation function");
+                return 0;
+        }
+    }
+
+    //logistic function
+    private double sigmoid(double x) {
+        return(1.0d / (1.0d + Math.exp(-x)));
+    }
+
+    //logistic function derivative
+    private double sigmoid_derivative(double x){
+        return (sigmoid(x)*(1.0d - sigmoid(x)));
+    }
+
+    //hyperbolic tangent
+    private double tanh(double x) {
+        return((1.0d - Math.exp(-2.0d*x))/(1.0d + Math.exp(-2.0d*x)));
+    }
+
+    //hyperbolic tangent derivative
+    private double tanh_derivative(double x) {
+        return (1.0d - Math.pow(tanh(x),2.0d));
+    }
+
+    //rectified linear unit
+    private double relu(double x) {
+        return (Math.max(x,0.0d));
+    }
+
+    //rectified linear unit derivative
+    private double relu_derivative(double x) {
+        if (x > 0) return 1.0f;
+        else if (x == 0) return 0;
+        else return 0;
+    }
+
+    //leaky rectified linear unit
+    private double leaky_relu(double x) {
+        if (x > 0) return x;
+        else return (0.01d*x);
+    }
+
+    //leaky rectified linear unit derivative
+    private double leaky_relu_derivative(double x) {
+        if (x > 0) return 1.0d;
+        else if (x == 0) return 0;
+        else return 0.01d;
+    }
+
+    //softplus function
+    private double softplus(double x) {
+        return  Math.log(1.0d + Math.exp(x));
+    }
+
+    //softplus function derivative
+    private double softplus_derivative(double x) {
+        return sigmoid(x);
+    }
+
+    public ScoreSet scores(Object example){
+        return null;
+    }
+
+    public ScoreSet scores(int[] exampleFeatures, double[] exampleValues) {
+        return null;
+    }
+
+    public void write(ExceptionlessOutputStream out) {
+        out.close();
```

Comment:
Can you add proper TODO message here, so that we remember to implement it? 
To be honest classifiers are not that useful in Saul, if their serialization/deserialization is not working. 

FYI @kordjamshidi